api/manifest/manifest.go

```diff
@@ -682,16 +688,38 @@ func (me ManifestEntry) runAsync(m *Manifest, prefix, app, process string, cache
 		}
 	}
 
+	gateway, err := getDockerGateway()
+
+	if err != nil {
+		ch <- err
+		return
+	}
+
+	host := ""
+	container := ""
+
 	for _, port := range ports {
 		switch len(strings.Split(port, ":")) {
 		case 1:
-			args = append(args, "-p", fmt.Sprintf("%s:%s", port, port))
+			host = port
+			container = port
 		case 2:
-			args = append(args, "-p", port)
+			parts := strings.SplitN(port, ":", 2)
+			host = parts[0]
+			container = parts[1]
 		default:
 			ch <- fmt.Errorf("unknown port declaration: %s", port)
 			return
 		}
+
+		switch me.Label(fmt.Sprintf("com.convox.port.%s.protocol", container)) {
+		case "proxy":
+			rnd := RandomPort()
+			go proxyPort(host, fmt.Sprintf("%s:%d", gateway, rnd))
```

Comment:
proxyPort already calls a go routine,
I think we should take the go routine out of proxyPort. and call it asynchronously here as that is very indicative of intent